### PR TITLE
feat: add support to filter skills by names

### PR DESCRIPTION
**Pull request by muneebGH**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [ ] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: taxonomy/api/v1/views.py [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
+from django_filters.rest_framework import DjangoFilterBackend
 from rest_framework import permissions
 from rest_framework.mixins import ListModelMixin, RetrieveModelMixin
 from rest_framework.viewsets import GenericViewSet, ModelViewSet
 
-from django_filters.rest_framework import DjangoFilterBackend
 from django.db.models import Prefetch
 
+from taxonomy.api.filters import SkillNameFilter
+from taxonomy.api.permissions import IsOwner
```
Comment (saleem-latif) [inline taxonomy/api/v1/views.py:12]: I do not see any use of `IsOwner` in the PR, please remove if this import is not being used anywhere.

**Review by saleem-latif — APPROVED** [review]
One comment, once that is addressed 👍🏼

File: taxonomy/api/filters.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+"""
+
+from django_filters import rest_framework as filters
+
+from taxonomy.models import Skill
+
+
+class SkillNameFilter(filters.FilterSet):
+    """
+    Filter skills by name. Supports filtering by comma-delimited list of names.
```
Comment (muhammad-ammar) [inline taxonomy/api/filters.py:12]: `list of names` -> `string of names`?

**Review by muhammad-ammar — APPROVED** [review]

File: taxonomy/api/v1/views.py (hunk 1 above)
Comment (muneebGH) [reply taxonomy/api/v1/views.py:12]: it is being used [here](https://github.com/openedx/taxonomy-connector/blob/e3af609230fc6776ade4fa216bf44d674ccabec2/taxonomy/api/v1/views.py#L86)
I just isorted the file

**Review by muhammad-ammar — APPROVED** [review]

**Pull request merged**